zebrad/tests/acceptance.rs

```diff
@@ -476,3 +476,15 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+#[ignore]
+fn sync_one_checkpoint() -> Result<()> {
+    zebra_test::init();
+
+    tempdir(ConfigMode::Persistent)?
+        .spawn_child(&["start"])?
+        .with_timeout(Duration::from_secs(1))
+        .expect_stdout("verified checkpoint range")?
+        .kill()
```

Comment:
I like this new API!

We want to do networking tests for mainnet and testnet, so it would be great to have a function that sets the network based on a zebra_chain::Network value.

Reply:
hmm, i looked around to see where we mess with the Network for tests and it seems to be only ever as part of the config. As such I'm thinking it might be better to rework how we pass around / in a config to make it easier to manipulate the Network. I'll push up a version of what I mean rn.

Reply:
Most of the app just uses the `config.network.network` value to make config-related decisions.

The one exception is `config.network.listener`, where the default port should change for testnet. In this case, we just issue a warning, because using 8233 on the testnet is a strange but legitimate config.

Reply:
sure, but this could all be part of the API on `Config` rather than the API on the test runners.

For example:

```
testdir()?
    .with_config(default_test_config().with_network(Testnet))
    .spawn_child(&["start"])?
    .with_timeout(Duration::from_secs(1))
    .expect_stdout("verified checkpoint range")?
    .kill()
```

then the `.with_network` function could even handle the listener issue you mentioned.

Reply:
Sounds good!